src/test/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/unit/service/DomainEventWorkerTest.kt

```diff
@@ -0,0 +1,314 @@
+package uk.gov.justice.digital.hmpps.approvedpremisesapi.unit.service
+
+import com.amazonaws.services.sns.model.MessageAttributeValue
+import com.amazonaws.services.sns.model.PublishRequest
+import com.amazonaws.services.sns.model.PublishResult
+import com.fasterxml.jackson.databind.ObjectMapper
+import com.fasterxml.jackson.datatype.jdk8.Jdk8Module
+import com.fasterxml.jackson.datatype.jsr310.JavaTimeModule
+import com.fasterxml.jackson.module.kotlin.registerKotlinModule
+import io.mockk.every
+import io.mockk.mockk
+import io.mockk.verify
+import org.junit.jupiter.api.Test
+import org.springframework.retry.backoff.FixedBackOffPolicy
+import org.springframework.retry.policy.SimpleRetryPolicy
+import org.springframework.retry.support.RetryTemplate
+import uk.gov.justice.digital.hmpps.approvedpremisesapi.api.events.model.ApplicationAssessedEnvelope
+import uk.gov.justice.digital.hmpps.approvedpremisesapi.api.events.model.ApplicationSubmittedEnvelope
+import uk.gov.justice.digital.hmpps.approvedpremisesapi.factory.events.ApplicationAssessedFactory
+import uk.gov.justice.digital.hmpps.approvedpremisesapi.factory.events.ApplicationSubmittedFactory
+import uk.gov.justice.digital.hmpps.approvedpremisesapi.jpa.entity.DomainEventEntity
+import uk.gov.justice.digital.hmpps.approvedpremisesapi.jpa.entity.DomainEventRepository
+import uk.gov.justice.digital.hmpps.approvedpremisesapi.model.DomainEvent
+import uk.gov.justice.digital.hmpps.approvedpremisesapi.model.domainevent.SnsEvent
+import uk.gov.justice.digital.hmpps.approvedpremisesapi.model.domainevent.SnsEventAdditionalInformation
+import uk.gov.justice.digital.hmpps.approvedpremisesapi.model.domainevent.SnsEventPersonReference
+import uk.gov.justice.digital.hmpps.approvedpremisesapi.model.domainevent.SnsEventPersonReferenceCollection
+import uk.gov.justice.digital.hmpps.approvedpremisesapi.service.AsyncDomainEventWorker
+import uk.gov.justice.digital.hmpps.approvedpremisesapi.service.SyncDomainEventWorker
+import uk.gov.justice.hmpps.sqs.HmppsQueueService
+import uk.gov.justice.hmpps.sqs.HmppsTopic
+import java.time.Instant
+import java.time.OffsetDateTime
+import java.time.ZoneOffset
+import java.util.UUID
+
+@Suppress("SwallowedException")
+class DomainEventWorkerTest {
+  private val domainEventRespositoryMock = mockk<DomainEventRepository>()
+  private val hmppsQueueServieMock = mockk<HmppsQueueService>()
+  private val objectMapper = ObjectMapper().apply {
+    registerModule(Jdk8Module())
+    registerModule(JavaTimeModule())
+    registerKotlinModule()
+  }
+  private val hmppsTopicMock = mockk<HmppsTopic>()
+
+  private val asyncDomainEventWorker = AsyncDomainEventWorker(
+    domainTopic = hmppsTopicMock,
+  )
+
+  private val syncDomainEventWorker = SyncDomainEventWorker(
+    domainTopic = hmppsTopicMock,
+  )
```

Comment:
I think it'd be worth initializing the `asyncDomainEventWorker` and `syncDomainEventWorker` using the `ConfiguredDomainEventWorker` interface, so we can test that works as expected:

```
private val asyncDomainEventWorker = ConfiguredDomainEventWorker(
  hmppsQueueService,
  false
)

private val syncDomainEventWorker = ConfiguredDomainEventWorker(
  hmppsQueueService,
  true
)
```

We could then mock the `hmppsQueueServiceMock` in a beforeEach like so:

```
@BeforeEach
fun setup() {
   every { hmppsQueueServieMock.findByTopicId("domainevents") } returns asyncDomainEventWorker.domainTopic
}
```